ArduPlane/mode_qrtl.cpp

```diff
@@ -3,11 +3,46 @@
 
 #if HAL_QUADPLANE_ENABLED
 
+bool ModeQRTL::enter_should_just_qland() const
+{
+    if (plane.quadplane.guided_wait_takeoff_on_mode_enter) {
+        return true;
+    }
+
+    if (plane.previous_mode->is_vtol_mode()) {
```

Comment:
I think either is_flying or some ground speed threshold would be worthwhile, or calculate the destination for the QRTL fixed wing flight and see if we are within 50m of that?
the problem with the spool change, especially get_spool_state(), is we could be at cruise speed when this triggers, and could be 50km away over a lake
@IamPete1 comments?